### PR TITLE
Add optional breadcrumbs tracking to calls/mutations of params

### PR DESCRIPTION
**Pull request by kyleboe**

### CODE REVIEW

File: lib/moderate_parameters/breadcrumbs.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      internal_param_logging(key, 'read', caller_locations) if ModerateParameters.breadcrumbs_enabled
+      super
+    end
+
+    def []=(key, value)
+      internal_param_logging(key, 'overwritten', caller_locations) if ModerateParameters.breadcrumbs_enabled
+      super
+    end
+
+    def extract!(*keys)
```
Comment (benjaminwood) [inline lib/moderate_parameters/breadcrumbs.rb]: Are there plans to add additional methods that are deprecated?

**Comment by tjchambers** [comment]
IMO this could be helpful in a client engagement. Can we get this PR merged?

**Comment by nvick** [comment]
I will work on a separate PR for Appraisal.

File: lib/moderate_parameters/breadcrumbs.rb (hunk 1 above)
Comment (kyleboe) [reply lib/moderate_parameters/breadcrumbs.rb]: Yes

**Pull request merged**